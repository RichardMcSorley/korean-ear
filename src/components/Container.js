import React from "react";

const styles={
          container:{
            height: 'calc(100vh - 64px)',
          //  background: 'linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) ), url(assets/images/bg_1.jpg)',
            backgroundSize: 'cover',
            display: "flex",
            flexDirection: 'column',
            alignItems: 'center',
            padding: '32px',
            paddingTop: '16px',
            overflow: 'scroll'
          }
}

export default (props) => (
      <div style={styles.container}>
        {props.children}
      </div>
);
