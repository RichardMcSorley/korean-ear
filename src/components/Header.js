import React from "react";
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import BookIcon from 'material-ui/svg-icons/action/book';

const styles= {

    contentWrapper: {
          padding: '16px',
          margin: '16px',
          paddingTop: '64px'
        },
      }
export default (props) => (
  <div>
    <AppBar
      title="Korean Ear"
      zDepth={3}
      className="page-header"
      iconElementLeft={<IconButton><BookIcon /></IconButton>}
      iconElementRight={
        <IconMenu
            iconButtonElement={
              <IconButton><MoreVertIcon /></IconButton>
            }
            targetOrigin={{horizontal: 'right', vertical: 'top'}}
            anchorOrigin={{horizontal: 'right', vertical: 'top'}}
          >
            <MenuItem primaryText="Help" />
        </IconMenu>}
    />
    <div style={styles.contentWrapper}>
      {props.children}
    </div>

  </div>

);
