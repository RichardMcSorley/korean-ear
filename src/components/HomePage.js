import { connect } from "react-redux";
import PropTypes from "prop-types";
import React, { Component } from "react";

export const HomePage = (props) => (
  <div>
    HomePage
  </div>
);

export default connect()(HomePage);
