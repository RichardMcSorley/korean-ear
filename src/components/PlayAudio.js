import React from 'react';
import Sound from 'react-sound';

export default class PlayAudio extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      playPosition: 1,
      position: 0,
      volume: 100,
      loop: false,
      playStatus: Sound.status.PLAYING
    };
  }
handleSongFinishedPlaying = () =>{
  if(this.state.playPosition < this.props.playTimes)
  this.setState((prev)=>({playPosition: prev.playPosition + 1}))
}
  render() {
    const { volume, loop } = this.state;

    return (
      <div>
          <Sound
            url={'assets/audio/'+ this.props.folder + '/' + this.props.filename}
            autoLoad={true}
            playStatus={this.state.playStatus}
            playFromPosition={300}
            onFinishedPlaying={this.handleSongFinishedPlaying}
          />
      </div>
    );
  }

}
