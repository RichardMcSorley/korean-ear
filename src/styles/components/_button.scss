.button {
  background-color: $blue;
  color: white;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
  border: none;
  padding: $s-size;
  font-weight: 300;
  font-size: $font-size-large;
}

.button--link{
  background: none;
}

.button--secondary {
  background: #888888;
}
.button--product{
  div{
    span{
      text-transform: none!important;
      font-weight: 400!important;
      font-size: 21px !important;
    }
  }
}
.button--skinny{
  div{
    span{
      padding: 0!important;

    }
  }

}
.button--skinny-outline{
  div{
    span{
      padding: 0!important;
      margin: 0!important;
      width: 0 !important;
      height: 0 !important;
    }
  }
}
