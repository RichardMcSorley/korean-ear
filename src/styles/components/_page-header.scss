.page-header{
  position: fixed!important;
  top: 0
}
