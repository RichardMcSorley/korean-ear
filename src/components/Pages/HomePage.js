import { connect } from "react-redux";
import React from "react";
import {update} from '../../actions/selection'
import FlatButton from 'material-ui/FlatButton'
import Header from '../Header'
import { history } from '../../routers/AppRouter'

const styles={

          button: {
            margin: '10px'
          }
}
const handleGoto = (multiplier, practice, props) =>{

  props.update({multiplier, practice})
  history.push('/select')

}
export const HomePage = (props) => (

  <div>
    <Header>
      <div>
        <FlatButton style={styles.button} onClick={()=>handleGoto(0, true, props)} backgroundColor="#0094FF" rippleColor="" label="연습 PRACTICE" fullWidth={true} />
        <FlatButton style={styles.button} onClick={()=>handleGoto(3, false, props)} backgroundColor="#F00" rippleColor="" label="초 BEGINNER" fullWidth={true} />
        <FlatButton style={styles.button} onClick={()=>handleGoto(2, false, props)} backgroundColor="#FFD800" rippleColor="" label="중 INTERMEDIATE" fullWidth={true} />
        <FlatButton style={styles.button} onClick={()=>handleGoto(1, false, props)} backgroundColor="#B6FF00" rippleColor="" label="고 ADVANCED" fullWidth={true} />

      </div>
  </Header>


  </div>
);

const mapDispatchToProps = ((dispatch)=>{
  return {
    update: (selection)=>dispatch(update(selection))
  }
})

export default connect(undefined, mapDispatchToProps)(HomePage);
