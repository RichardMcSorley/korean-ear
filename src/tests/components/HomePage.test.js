import { shallow } from "enzyme";
import {HomePage} from "../../components/HomePage";
import React from "react";

test("should render HomePage correctly", () => {
  const wrapper = shallow(<HomePage />);
  expect(wrapper).toMatchSnapshot();
});
