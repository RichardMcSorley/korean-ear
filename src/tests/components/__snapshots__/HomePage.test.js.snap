// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage correctly 1`] = `
<div
  style={
    Object {
      "alignItem": "center",
      "background": "grey",
      "backgroundSize": "cover",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "width": "100vw",
    }
  }
>
  <div>
    <h1>
      Tobacco
    </h1>
    <p>
      Tagline
    </p>
    <Link
      replace={false}
      to="/products"
    >
      Start
    </Link>
  </div>
</div>
`;
